snorkel/analysis/utils.py

```diff
@@ -0,0 +1,251 @@
+    -------
+    np.ndarray
+        The input converted to an np.ndarray
+
+    Raises
+    ------
+    ValueError
+        Provided input could not be converted to an np.ndarray
+    """
+
+    orig_type = type(array_like)
+
+    # Convert to np.ndarray
+    if isinstance(array_like, np.ndarray):
+        pass
+    elif isinstance(array_like, list):
+        array_like = np.array(array_like)
+    elif isinstance(array_like, sparse.spmatrix):
+        array_like = array_like.toarray()
+    elif isinstance(array_like, torch.Tensor):
+        array_like = array_like.numpy()
+    elif not isinstance(array_like, np.ndarray):
+        array_like = np.array(array_like)
+    else:
+        msg = f"Input of type {orig_type} could not be converted to an np.ndarray"
+        raise ValueError(msg)
+
+    # Correct shape
+    if flatten:
+        if (array_like.ndim > 1) and (1 in array_like.shape):
+            array_like = array_like.flatten()
+        if array_like.ndim != 1:
+            raise ValueError("Input could not be converted to 1d np.array")
+
+    # Convert to ints
+    if cast_to_int:
+        if any(array_like % 1):
+            raise ValueError("Input contains at least one non-integer value.")
+        array_like = array_like.astype(np.dtype(int))
+
+    return array_like
+
+
+def convert_labels(Y: ArrayLike, source: str, target: str) -> ArrayLike:
+    """Convert a matrix from one label type to another
+
+    Args:
+        Y: A np.ndarray or torch.Tensor of labels (ints) using source convention
+        source: The convention the labels are currently expressed in
+        target: The convention to convert the labels to
+    Returns:
+        Y: an np.ndarray or torch.Tensor of labels (ints) using the target convention
+
+    Conventions:
+        'categorical': [0: abstain, 1: positive, 2: negative]
+        'plusminus': [0: abstain, 1: positive, -1: negative]
+        'onezero': [0: negative, 1: positive]
+
+    Note that converting to 'onezero' will combine abstain and negative labels.
+    """
+    if Y is None:
+        return Y
+    if isinstance(Y, np.ndarray):
+        Y = Y.copy()
+        assert Y.dtype == np.int64
+    elif isinstance(Y, torch.Tensor):
+        Y = Y.clone()
+        assert isinstance(Y, torch.LongTensor)
+    else:
+        raise ValueError("Unrecognized label data type.")
+    negative_map = {"categorical": 2, "plusminus": -1, "onezero": 0}
+    Y[Y == negative_map[source]] = negative_map[target]
+    return Y
+
+
+def plusminus_to_categorical(Y: ArrayLike) -> ArrayLike:
+    """Convert labels from plus-minus to categorical
+
+    Parameters
+    ----------
+    Y
+        Integer labels in plus-minus form plus-minus: (-1 = neg, 0 = abstain, +1 = pos)
+
+    Returns
+    -------
+    ArrayLike
+        Integer labels in categorical form categorical: (0 = abstain, 1 = pos, 2 = neg)
+    """
+    return convert_labels(Y, "plusminus", "categorical")
+
+
+def categorical_to_plusminus(Y: ArrayLike) -> ArrayLike:
```

Comment:
do we need these separate functions given they're a simple return statement? the `convert_labels` function is clear enough

Reply:
I vote we drop them too. We'll just need to pull out where they're used in the tutorials. I'll make a note for a subsequent PR.